### PR TITLE
Implements a map of postal offices

### PR DESCRIPTION
**Pull request by Bogovesov**
#Implemets #10
@internal-jew PTAL

### CODE REVIEW

**Comment by Bogovesov** [comment]
@internal-jew PTAL

File: logistics/src/main/java/com/alevel/deliverit/postal/network/Connection.java [hunk 1]
```diff
@@ -67,24 +69,39 @@ public Builder setEndNode(PostalUnit endNode) {
...
+            return new Connection(startNode, endNode, constraints);
         }
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof Connection)) return false;
+        Connection that = (Connection) o;
+        return Objects.equals(getStartNode(), that.getStartNode()) &&
```
Comment (internal-jew) [inline logistics/src/main/java/com/alevel/deliverit/postal/network/Connection.java]: Static import?

File: logistics/src/main/java/com/alevel/deliverit/postal/network/dijkstra/Dijkstra.java [hunk 2]
```diff
@@ -62,15 +64,26 @@ private void buildRoute(Set<Connection> connections) {
         }
     }
 
+    private void checkInitialNodeForVisit(Set<Connection> connections, Map<PostalUnit, Integer> localWeight) {
+        if ((currentRouteNumber == 0) && (nodeWeights.containsKey(endNode) && (nodesForVisit.isEmpty()))) {
```
Comment (internal-jew) [inline logistics/src/main/java/com/alevel/deliverit/postal/network/dijkstra/Dijkstra.java]: скобки

File: logistics/src/main/java/com/alevel/deliverit/postal/network/dijkstra/Dijkstra.java [hunk 3]
```diff
@@ -62,15 +64,26 @@ private void buildRoute(Set<Connection> connections) {
...
+                }
+            });
+        }
+    }
+
     private void getStartNodeFromSavedForVisit() {
         if (!nodesForVisit.isEmpty()) {
             final Connection connection = nodesForVisit.poll();
-            assert null != connection;
+            Preconditions.checkNotNull(connection);
```
Comment (internal-jew) [inline logistics/src/main/java/com/alevel/deliverit/postal/network/dijkstra/Dijkstra.java]: static import

File: logistics/src/main/java/com/alevel/deliverit/postal/network/dijkstra/Dijkstra.java [hunk 4]
```diff
@@ -103,41 +115,47 @@ private void searchNextStartNode(Map<PostalUnit, Integer> nodes) {
...
             int weight = nodeWeights.getOrDefault(endNode, 0);
-
-            if ((nodes.contains(endNode)) && (route.getWeight() == weight)) {
-                System.out.println("Route from '" + startNode.getName() + "' to '" + endNode.getName() + "'");
-
-                System.out.print("[" + startNode.getName());
-                nodes.forEach(node -> System.out.print(" -> " + node.getName()));
-                System.out.print("] weight: " + route.getWeight());
-                System.out.println();
+            if ( (nodes.contains(endNode)) && (route.getWeight() == weight)) {
```
Comment (internal-jew) [inline logistics/src/main/java/com/alevel/deliverit/postal/network/dijkstra/Dijkstra.java]: Скобки и форматирование.

File: logistics/src/main/java/com/alevel/deliverit/postal/network/dijkstra/Dijkstra.java [hunk 5]
```diff
@@ -165,27 +183,29 @@ public Builder setEndNode(PostalUnit endNode) {
             return this;
         }
 
-        public RouteMap build() {
+        public Dijkstra build() {
```
Comment (internal-jew) [inline logistics/src/main/java/com/alevel/deliverit/postal/network/dijkstra/Dijkstra.java]: `DijkstraAlogrithm`

**Comment by Bogovesov** [comment]
@internal-jew PTAL

File: logistics/src/main/java/com/alevel/deliverit/postal/network/dijkstra/DijkstraAlgorithm.java [hunk 6]
```diff
@@ -132,7 +135,7 @@ private Route getShortestRoute() {
         routes.forEach((num, route) -> {
             List<PostalUnit> nodes = route.getUnits();
             int weight = nodeWeights.getOrDefault(endNode, 0);
-            if ( (nodes.contains(endNode)) && (route.getWeight() == weight)) {
+            if ((nodes.contains(endNode)) && (route.getWeight() == weight)) {
```
Comment (internal-jew) [inline logistics/src/main/java/com/alevel/deliverit/postal/network/dijkstra/DijkstraAlgorithm.java]: To many brackets

File: logistics/src/main/java/com/alevel/deliverit/postal/network/dijkstra/DijkstraAlgorithm.java [hunk 7]
```diff
@@ -145,7 +148,7 @@ private void printRoutes(Route route) {
         List<PostalUnit> nodes = route.getUnits();
         int weight = nodeWeights.getOrDefault(endNode, 0);
 
-        if ((nodes.contains(endNode)) && (route.getWeight() == weight)) {
+        if (nodes.contains(endNode) && (route.getWeight() == weight)) {
```
Comment (internal-jew) [inline logistics/src/main/java/com/alevel/deliverit/postal/network/dijkstra/DijkstraAlgorithm.java]: Redundant brackets

**Pull request merged**